lib/BaseStrategy.ts

```diff
@@ -74,7 +74,7 @@ export abstract class BaseStrategy extends Strategy {
         return this.error(err)
       }
       if (!user) {
-        return self.fail(info)
+        return this.fail(info)
```

Comment:
Why this change? Was this just a bug that you noticed?

Reply:
Idk, why typescript did not report that, but there is not `self` variable. While I was debugging I once run into that if statement and it throws me runtime error. It's look like someone was thinking about coding in python ;)